app/main.py

```diff
@@ -494,9 +494,22 @@ def initialize(self, request, response):
 
     def serve(self):
         request, response, env = self.request, self.response, self.env
+
+        # If the Person Finder instance has not been initialized yet,
+        # prepend to any served page a warning and a link to initialize
+        # the datastore.
+        if 0 == len(env.repo_options) and env.action != 'setup_datastore':
+            get_vars = lambda: {'env': env}
+            content = resources.get_rendered('setup_datastore.html', env.lang,
+                    (env.repo, env.charset), get_vars)
+            response.out.write(content)
```

Comment:
In this case, I believe we can just show setup_datastore.html without the actual content i.e., "return" just after this line. Because there's nothing to do for uninitialized repositories other than initializing it. But you can leave it for now. I can change it later.